keras_nlp/metrics/rouge_base.py

```diff
@@ -0,0 +1,186 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""ROUGE metric implementation based on `keras.metrics.Metric`."""
+
+
+import tensorflow as tf
+from tensorflow import keras
+
+from keras_nlp.utils.tensor_utils import tensor_to_string_list
+
+try:
+    import rouge_score
+    from rouge_score import rouge_scorer
+except ImportError:
+    rouge_score = None
+
+
+class RougeBase(keras.metrics.Metric):
+    """ROUGE metric.
+    This class implements all the variants of the ROUGE metric - ROUGE-N,
+    ROUGE-L and ROUGE-LSum.
+    Args:
+        variant: string. One of "rougeN", "rougeL", "rougeLsum". Defaults to
+            "rouge2". For "rougeN", N lies in the range [1, 9].
+        metric_type: string. One of "precision", "recall", "f1_score". Defaults
+            to "f1_score".
+        use_stemmer: bool. Whether Porter Stemmer should be used to strip word
+            suffixes to improve matching. Defaults to False.
+        dtype: string or tf.dtypes.Dtype. Precision of metric computation. If
+               not specified, it defaults to tf.float32.
+        name: string. Name of the metric instance.
+        **kwargs: Other keyword arguments.
+    """
+
+    def __init__(
+        self,
+        variant="rouge2",
+        metric_type="f1_score",
```

Comment:
We still need to figure out why returning a dict is not working, and see if there is a bug that needs to be fixed in core keras or elsewhere.

We should not ship a API signature we don't want because of a bug we need to fix!